metricshub-agent/src/main/java/org/sentrysoftware/metricshub/agent/helper/ConnectorVariablesLibraryParser.java

```diff
@@ -0,0 +1,288 @@
+ * final Connector (with a displayName section), and using ConnectorParser to parse the YAML file and create
+ * Connector objects.
+ * </p>
+ */
+@Slf4j
+public class ConnectorVariablesLibraryParser {
+
+	/**
+	 * This inner class allows to visit the files contained within the Yaml directory
+	 */
+	private static class ConnectorFileVisitor extends SimpleFileVisitor<Path> {
+
+		private final Map<String, AdditionalConnector> additionalConnectorConfig;
+		private final AdditionalConnectorsParsingResult connectorsParsingResult = new AdditionalConnectorsParsingResult();
+
+		public AdditionalConnectorsParsingResult getConnectorsParsingResult() {
+			return connectorsParsingResult;
+		}
+
+		ConnectorFileVisitor(final Map<String, AdditionalConnector> additionalConnectorConfig) {
+			this.additionalConnectorConfig = additionalConnectorConfig;
+		}
+
+		@Override
+		public FileVisitResult visitFile(Path path, BasicFileAttributes attrs) throws IOException {
+			// Skip this path if it is a directory or not a YAML file
+			if (Files.isDirectory(path) || !isYamlFile(path.toFile().getName())) {
+				return FileVisitResult.CONTINUE;
+			}
+			final ObjectMapper yamlMapper = new YAMLMapper();
+			final JsonNode connectorNode = yamlMapper.readTree(path.toFile());
+			if (!isConnector(connectorNode)) {
+				return FileVisitResult.CONTINUE;
+			}
+
+			// Get the connector's file name
+			final String filename = path.getFileName().toString();
+			final String connectorId = filename.substring(0, filename.lastIndexOf('.'));
+
+			if (!connectorNode.toString().contains("${var::")) {
+				return FileVisitResult.CONTINUE;
+			}
+
+			// Normalize additionalConnectors
+			normalizeAdditionalConnectors();
+
+			// Removing all the configurations that are not using this connector.
+			final Map<String, AdditionalConnector> filteredConnectors = additionalConnectorConfig
+				.entrySet()
+				.stream()
+				.filter(entry -> connectorId.equalsIgnoreCase(entry.getValue().getUses()))
+				.collect(
+					Collectors.toMap(
+						Map.Entry::getKey, // Keep the original key
+						Map.Entry::getValue // Keep the original AdditionalConnectorConfig as the value
+					)
+				);
+
+			// Construct a variables map from the default connector variables.
+			final Map<String, String> defaultVariables = new HashMap<>(getDefaultConnectorVariables(connectorNode));
+
+			// Parse the connector even if it is not configured as an additional connector.
+			// This ensures that the connector will function with the default variables if the user forces it.
+			if (filteredConnectors.isEmpty()) {
+				parseConnectorWithModifier(path, defaultVariables, connectorId, filename, connector -> {});
+			}
+
+			// For each configuration, we create a new custom connector and a new variables map to be used in the connector update.
+			for (final Entry<String, AdditionalConnector> connectorConfigurationEntry : filteredConnectors.entrySet()) {
```

Comment:
We shouldn't call this loop if `filteredConnectors.isEmpty()`. Move the code of this loop in a new method and use an if else statement.
```java
if (filteredConnectors.isEmpty()) {
	parseConnectorWithModifier(path, defaultVariables, connectorId, filename, Optional.empty());
} else {
        generateNewConnectors(...)
}